content/docs/security/ratelimit/global.md

```diff
@@ -0,0 +1,432 @@
+  - key: service
+    value: premium-api
+    rate_limit:
+      unit: minute
+      requests_per_unit: 2
+```
+
+## Step 3: Create a GatewayExtension {#gateway-extension}
+
+Create a GatewayExtension resource that points to your Rate Limit Service. 
+
+1. Create a GatewayExtension resource that points to your Rate Limit Service.
+
+   ```yaml
+   kubectl apply -f - <<EOF
+   apiVersion: gateway.kgateway.dev/v1alpha1
+   kind: GatewayExtension
+   metadata:
+     namespace: {{< reuse "docs/snippets/namespace.md" >}}
+     name: global-ratelimit
+   spec:
+     type: RateLimit
+     rateLimit:
+       grpcService:
+         backendRef:
+           name: ratelimit
+           namespace: kgateway-test
+           port: 8081
+       domain: "api-gateway"
+       timeout: "100ms"
+       failOpen: false
+   EOF
+   ```
+
+   {{% reuse "docs/snippets/review-table.md" %}} For more information, see the [API docs](../../reference/api/#ratelimitprovider).
+
+   | Field | Description | Required |
+   |-------|-------------|----------|
+   | grpcService | Configuration for connecting to the gRPC rate limit service. | Yes |
+   | domain | Domain identity for the rate limit service. | Yes |
+   | timeout | Timeout for rate limit service calls, such as `100ms`. | No |
+   | failOpen | When `true`, requests continue even if the rate limit service is unavailable. | No (defaults to `false`) |
+
+2. Create a Kubernetes ReferenceGrant to allow the GatewayExtension to access the Rate Limit Service. Otherwise, you can create the GatewayExtension and Gateway in the same namespace as the Rate Limit Service.
+
+   ```yaml
+   kubectl apply -f - <<EOF
+   apiVersion: gateway.networking.k8s.io/v1beta1
+   kind: ReferenceGrant
+   metadata:
+     name: global-ratelimit
+     namespace: kgateway-test
+   spec:
+     from:
+     - group: gateway.kgateway.dev
+       kind: GatewayExtension
+       namespace: {{< reuse "docs/snippets/namespace.md" >}}
+     to:
+     - group: ""
+       kind: Service
+   EOF
+   ```
+
+## Step 4: Create a TrafficPolicy {#traffic-policy}
+
+Create a TrafficPolicy resource that applies rate limits to your routes. 
+
+Flip through the tabs for different example rate limit policies. Note that the examples apply to the Gateway that you created before you began, but you can also apply a TrafficPolicy to an HTTPRoute or specific route.
+
+{{< tabs tabTotal="5" items="Client IP address, Request path, User ID, Combined rate limiting, Combined local and global rate limiting" >}}
+
+{{% tab tabName="Client IP address" %}}
+
+Limit requests based on the client's IP address.
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.kgateway.dev/v1alpha1
+kind: TrafficPolicy
+metadata:
+  name: ip-rate-limit
+  namespace: {{< reuse "docs/snippets/namespace.md" >}}
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: http
+  rateLimit:
+    global:
+      descriptors:
+      - entries:
+        - type: RemoteAddress
```

Comment:
does this need to match the descriptor in the rate limit service? if so, the RLS uses `remote_address`